lib/mix/tasks/gen.migration.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Mix.Tasks.Gen.Migration do
+  @moduledoc """
+  Generates a new migration file.
+  """
+  use Mix.Task
+
+  import Macro, only: [camelize: 1, underscore: 1]
+  import Mix.Generator
+
+  @impl true
+  def run(args) do
+    case OptionParser.parse!(args, strict: []) do
+      {_opts, [name]} ->
+        path = Path.join(:code.priv_dir(:ae_mdw), "migrations")
+        base_name = "#{underscore(name)}.ex"
+        file = Path.join(path, "#{timestamp()}_#{base_name}")
+        unless File.dir?(path), do: create_directory(path)
+
+        fuzzy_path = Path.join(path, "*_#{base_name}")
+
+        if Path.wildcard(fuzzy_path) != [] do
+          Mix.raise(
+            "migration can't be created, there is already a migration file with name #{name}."
+          )
+        end
+
+        # The :change option may be used by other tasks but not the CLI
+        assigns = [
+          # credo:disable-for-next-line
```

Comment:
Let's avoid skipping credo errors. We're using them for a reason

Reply:
If we use `safe_concat` here (as it suggests) it fails, because the atom is not yet created. The warning is specifically against dynamically creating atoms - but this is exactly what is wanted here.